driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -824,8 +824,7 @@ private int parseInteger(final String input, final String key) {
             if (host.length() == 0) {
                 throw new IllegalArgumentException(format("The connection string contains an empty host '%s'. ", rawHosts));
             } else if (host.endsWith(".sock")) {
-                throw new IllegalArgumentException(format("The connection string contains an invalid host '%s'. "
-                        + "Unix Domain Socket which is not supported by the Java driver", host));
+                host = urldecode(host);
```

Comment:
Add an example to the Javadoc (here and MongoClientURI) documenting that it supports .sock now.

Reply:
Done.